internal/api/utils_test.go

```diff
@@ -0,0 +1,83 @@
+package api
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/labstack/echo/v4"
+)
+
+func TestMapQueryParameters(t *testing.T) {
+	e := echo.New()
+	req := httptest.NewRequest(http.MethodGet, "/", nil)
+	rec := httptest.NewRecorder()
+	c := e.NewContext(req, rec)
+	now := time.Now().UTC()
+	firstOfMonth := time.Date(now.Year(), now.Month(), 1, 0, 0, 0, 0, time.UTC).Format(timeLayout)
+
+	type tests struct {
+		name     string
+		qinputs  map[string]string
+		qoutputs map[string][]string
+		errmsg   string
+	}
+
+	var all_tests = []tests{
+		{
+			name:    "When only start date is provided",
+			qinputs: map[string]string{"start_date": "2023-03-23", "end_date": ""},
+			qoutputs: map[string][]string{
+				"DATE(recommendation_sets.monitoring_end_time) >= ?": {"2023-03-23"},
+				"DATE(recommendation_sets.monitoring_end_time) <= ?": {now.Format(timeLayout)},
+			},
+			errmsg: "The start date should be greater than or equal to recommendation_sets.monitoring_end_time!",
+		},
+		{
+			name:    "When only end date is provided",
+			qinputs: map[string]string{"start_date": "", "end_date": "2023-03-24"},
+			qoutputs: map[string][]string{
+				"DATE(recommendation_sets.monitoring_end_time) <= ?": {"2023-03-24"},
+				"DATE(recommendation_sets.monitoring_end_time) >= ?": {firstOfMonth},
+			},
+			errmsg: "The recommendation_sets.monitoring_end_time should be less than or equal to end date!",
+		},
+		{
+			name:    "When start date and end date are provided",
+			qinputs: map[string]string{"start_date": "2023-03-23", "end_date": "2023-03-24"},
+			qoutputs: map[string][]string{
+				"DATE(recommendation_sets.monitoring_end_time) <= ?": {"2023-03-24"},
+				"DATE(recommendation_sets.monitoring_end_time) >= ?": {"2023-03-23"},
+			},
+			errmsg: `The recommendation_sets.monitoring_end_time should be less than or equal to end date!
+				The recommendation_sets.monitoring_end_time should be greater than or equal to start date!`,
+		},
+		{
+			name:    "When start date and end date are not provided",
```

Comment:
Regarding the first two test cases default values should kick in, additionally I don't see a test case where-in we are checking if start date is compared against monitoring_end_time instead of monitoring_start_time.
Are we trying to gather that from the json error response?

Reply:
@upadhyeammit As discussed, we will need a setup and teardown system to test the change in this PR, that said we can keep the below test in the highlighted section.

`name: "When start date and end date are provided"`